### PR TITLE
Implementation of duplication in clds

### PR DESCRIPTION
**Pull request by elizabeth-legros**
The purpose of this branch is to create a small pr to merge into staging of duplicating a data branch associated with a given node.  Duplicate multi is also part of this pr, utilizing the newly added logic

### CODE REVIEW

File: com.unity.sg2/Editor/ContextLayeredDataStorage/Element.cs [hunk 1]
```diff
@@ -61,6 +62,18 @@ internal virtual SerializedElementData ToSerializedFormat()
             return new SerializedElementData(ID, Header.GetType().AssemblyQualifiedName, Header.ToJson(), null, null);
         }
 
+        public virtual Element MakeCopy()
```
Comment (brett-groshong) [inline com.unity.sg2/Editor/ContextLayeredDataStorage/Element.cs:65]: Should this be a `Clone` method override?

File: com.unity.sg2/Editor/ContextLayeredDataStorage/Header.cs [hunk 2]
```diff
@@ -24,5 +24,10 @@ public virtual void FromJson(string json)
         {
 
         }
+
+        public virtual DataHeader MakeCopy()
+        {
+            return new DataHeader();
```
Comment (brett-groshong) [inline com.unity.sg2/Editor/ContextLayeredDataStorage/Header.cs:30]: Copies are always empty?

**Review by brett-groshong — APPROVED** [review]

File: com.unity.sg2/Editor/ContextLayeredDataStorage/Header.cs (hunk 2 above)
Comment (elizabeth-legros) [reply com.unity.sg2/Editor/ContextLayeredDataStorage/Header.cs:30]: in the base case, yes

File: com.unity.sg2/Editor/ContextLayeredDataStorage/Element.cs (hunk 1 above)
Comment (elizabeth-legros) [reply com.unity.sg2/Editor/ContextLayeredDataStorage/Element.cs:65]: Oh is that a c# convention I should follow? I wasnt familiar with it, but I can make that change if you think I should

Comment (brett-groshong) [reply com.unity.sg2/Editor/ContextLayeredDataStorage/Element.cs:65]: It appears that the convention is to have copy constructors. I'm not sure if we should standardize on this. We can take this to a Slack thread and I will resolve this convo.

**Review by brett-groshong — APPROVED** [review]

**Pull request merged**